broker/lib/controllers/ServiceBrokerApiController.js

```diff
@@ -72,7 +83,64 @@ class ServiceBrokerApiController extends FabrikBaseController {
       .catch(ServiceInstanceAlreadyExists, conflict);
   }
 
+  putInstanceV2(req, res) {
+    const params = _.omit(req.body, 'plan_id', 'service_id');
+
+    function done(result) {
+      let statusCode = CONST.HTTP_STATUS_CODE.CREATED;
+      const body = {
+        dashboard_url: req.instance.dashboardUrl
+      };
+      if (req.instance.async) {
+        statusCode = CONST.HTTP_STATUS_CODE.ACCEPTED;
+        body.operation = utils.encodeBase64(result);
+      }
+      res.status(statusCode).send(body);
+    }
+
+    function conflict(err) {
+      /* jshint unused:false */
+      res.status(CONST.HTTP_STATUS_CODE.CONFLICT).send({});
+    }
+
+    req.operation_type = CONST.OPERATION_TYPE.CREATE;
+    this.validateRequest(req, res);
+
+    return Promise.try(() => {
+        if (req.manager.name === CONST.INSTANCE_TYPE.DIRECTOR) {
+          // Acquire lock for this instance
+          return lockManager.lock(req.params.instance_id, {
+            lockType: CONST.ETCD.LOCK_TYPE.WRITE,
+            lockedResourceDetails: {
+              resourceType: CONST.APISERVER.RESOURCE_TYPES.DEPLOYMENT,
+              resourceName: CONST.APISERVER.RESOURCE_NAMES.DIRECTOR,
+              resourceId: req.params.instance_id,
+              operation: CONST.OPERATION_TYPE.CREATE
+            }
+          });
+        }
+      })
+      .then(() => req.instance.create(params))
+      .then(done)
+      // Release lock in case of error: catch and throw
+      .catch(err => {
+        if (err instanceof EtcdLockError) {
```

Comment:
Maybe instead of if and instanceof you can use [filtered-catch](http://bluebirdjs.com/docs/api/catch.html#filtered-catch) or [catchthrow](http://bluebirdjs.com/docs/api/catchthrow.html). Same for catch blocks below [#L225](https://github.com/cloudfoundry-incubator/service-fabrik-broker/pull/304/files#diff-03e33ae33c13c68e445aa8cc1048acc9R225), [#L305](https://github.com/cloudfoundry-incubator/service-fabrik-broker/pull/304/files#diff-03e33ae33c13c68e445aa8cc1048acc9R305), [#L411](https://github.com/cloudfoundry-incubator/service-fabrik-broker/pull/304/files#diff-03e33ae33c13c68e445aa8cc1048acc9R411) and [#L160](https://github.com/cloudfoundry-incubator/service-fabrik-broker/pull/304/files#diff-8d0f3cccd3d5ff4a6f6d89f5d44aceefR160) in BnRStatusPollerJob.js.

Reply:
What this piece of code does is to unlock resource only if thrown error is not EtcdLockError and throws the error.
Basically what we want here: catch all errors except EtcdLockError. catchthrow doesn't provide this functionality